src/wayscript/context.py

```diff
@@ -0,0 +1,56 @@
+import os
+from functools import lru_cache
+
+from . import utils
+
+
+@lru_cache()
+def _get_process_detail_expanded_data() -> dict:
+    """Wrapper to cache data from process_detail_expanded call"""
+    process_uuid = utils.get_process_uuid()
+    client = utils.WayScriptClient()
+    response = client.get_process_detail_expanded(process_uuid)
+    response.raise_for_status()
+    return response.json()
+
+
+def get_process():
+    """
+    Return metadata about the current execution process.
+    """
+    data = _get_process_detail_expanded_data()["process"]
+    return data
+
+
+def get_event():
+    """Returns the event object of the execution context"""
+    data = _get_process_detail_expanded_data()["event"]
+    return data
+
+
+def get_lair_trigger():
+    """Returns information about the trigger that launched the event"""
+    data = _get_process_detail_expanded_data()["lair_trigger"]
+    return data
+
+
+def get_lair():
+    """Returns lair metadata"""
+    process = get_process()
+    lair_id = process["lair_id"]
+    client = utils.WayScriptClient()
+    response = client.get_lair_detail(lair_id)
+    response.raise_for_status()
+    data = response.json()
+    return data
+
+
+def get_workspace():
```

Comment:
Should we also add the ability to get info about the user executing the call? I think right now we're just defaulting that to the owner of the lair. wdyt?

Reply:
Yeah, can definitely add a method to call out for user detail data. I can add a ticket for it under the context epic.

Reply:
https://linear.app/wayscript/issue/GEN-246/support-retrieving-execution-user-metadata